internal/fwserver/attribute_plan_modification.go

```diff
@@ -122,6 +131,7 @@ func AttributeModifyPlan(ctx context.Context, a fwschema.Attribute, req tfsdk.Mo
 					Plan:          resp.Plan,
 					ProviderMeta:  req.ProviderMeta,
 					State:         req.State,
+					Private:       privateProviderData,
```

Comment:
Open question: I wonder if these should be using the _updated_ private state data from the `AttributeWithPlanModifiers` handling above?

I think we may want that to make it slightly less confusing. It can use the `resp.Private` in these places here and below.

Either which way, I think we should document the behavior in the `ModifyAttributePlanRequest`/`ModifyAttributePlanResponse` fields to say that they either the private state is always the original data when the schema-based plan modification began or that its updated as the logic traverses deeper into underlying attributes.

Reply:
One way to unit test this would be to have an attribute implementing nested attributes and a plan modifier that updates the parent attribute while one of the nested attributes has a plan modifier that is conditional on that update.